pkg/bufferdecoder/decoder.go

```diff
@@ -29,11 +31,61 @@ var ErrBufferTooShort = errors.New("can't read context from buffer: buffer too s
 // The protocol is specific between the Trace eBPF program and the Tracee-eBPF user space application.
 func New(rawBuffer []byte) *EbpfDecoder {
 	return &EbpfDecoder{
-		buffer: rawBuffer,
-		cursor: 0,
+		buffer:  rawBuffer,
+		cursor:  0,
+		present: newDataPresentor(),
 	}
 }
 
+func newDataPresentor() map[trace.DecodeAs]map[string]func(any) (any, error) {
+	// type default presentor
+	identity := func(a any) (any, error) {
+		return a, nil
+	}
+
+	present := map[trace.DecodeAs]map[string]func(any) (any, error){
+		trace.INT_T:  {},
+		trace.UINT_T: {},
+		trace.LONG_T: {},
+		trace.ULONG_T: {
+			"time.Time": func(a any) (any, error) {
+				argVal, ok := a.(uint64)
+				if !ok {
+					return nil, errfmt.Errorf("error presenting uint64 as time.Time, type received was %T", a)
+				}
+				return time.NsSinceEpochToTime(time.BootToEpochNS(argVal)), nil
+			},
+		},
+		trace.U16_T:        {},
+		trace.U8_T:         {},
+		trace.INT_ARR_2_T:  {},
+		trace.UINT64_ARR_T: {},
+		trace.POINTER_T:    {},
+		trace.BYTES_T:      {},
+		trace.STR_T:        {},
+		trace.STR_ARR_T:    {},
+		trace.SOCK_ADDR_T:  {},
+		trace.CRED_T:       {},
+		trace.TIMESPEC_T: {
+			// timespec is seconds+nano in float
+			"float64": identity,
+		},
+		trace.ARGS_ARR_T: {},
+		trace.BOOL_T:     {},
+		trace.FLOAT_T:    {},
+		trace.FLOAT64_T:  {},
+	}
+
+	// add an identity function per decode type for its string presentation and an undefined type
+	for decode := range present {
+		present[decode][decode.String()] = identity
+		// note(nadav.str): this allows defining data fields with a type - should we allow it?
+		present[decode][""] = identity
+	}
+
+	return present
+}
+
```

Comment:
I'm afraid performing this map lookup for every data field of every event may have performance implications. Did you check for that?

Reply:
I believe it can be converted to array, right @NDStrahilevitz?

Reply:
At the second level I think we would inevitably have a map, but the first layer could be converted to a slice.